src/memory.rs

```diff
@@ -50,6 +51,25 @@ impl Pointer {
             offset: 0,
         }
     }
+    pub fn is_aligned_to(&self, align: usize) -> bool {
```

Comment:
I think you can disregard this and just respond to my long comment on the PR. I didn't really grasp what was going on at first.